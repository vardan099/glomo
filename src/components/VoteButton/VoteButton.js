import React from 'react';
import PropTypes from "prop-types";

const VoteButton = (props) => {
    const {buttonText, clickHandler, voteType, gameId, type} = props;
    return (
        <button data-test="voteButton" className={`vote-button bg-${type}`} onClick={() => clickHandler(gameId, voteType)}>{buttonText}</button>
    )
};

VoteButton.propTypes = {
    buttonText: PropTypes.string,
    clickHandler: PropTypes.func,
    voteType: PropTypes.number,
    type: PropTypes.string,
    gameId: PropTypes.string
};


export default VoteButton;