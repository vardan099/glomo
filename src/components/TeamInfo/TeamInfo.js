import React from 'react';
import PropTypes from "prop-types";
import VoteButton from "../VoteButton/VoteButton";

const voteTypes = {
    "home": {'voteType':1,'buttonText':'Home Team'},
    "draw": {'voteType':2,'buttonText':'Draw'},
    "away": {'voteType':3,'buttonText':'Away Team'},
};

const TeamInfo = (props) => {
    const {teamName, teamLogo, voteHandler, objectId, type} = props;
    return (
        <div className={`sports-poll_${type}-team`}>
            <img className='team-logo' src={teamLogo} alt={`${type} team logo`}/>
            <h4>{teamName}</h4>
            <VoteButton
                buttonText={voteTypes[type].buttonText}
                clickHandler={voteHandler}
                voteType={voteTypes[type].voteType}
                gameId={objectId}
                type={type}
            />
        </div>
    )
};

TeamInfo.propTypes = {
    teamName: PropTypes.string,
    teamLogo: PropTypes.string,
    voteHandler: PropTypes.func,
    type: PropTypes.string,
    objectId: PropTypes.string
};


export default TeamInfo;