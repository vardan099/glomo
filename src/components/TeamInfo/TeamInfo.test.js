import React from 'react';
import { shallow } from 'enzyme';
import { checkProps } from '../../TestUtils/index';
import TeamInfo from './TeamInfo'
import VoteButton from '../VoteButton/VoteButton';
import HomeTeamLogo from '../../images/home-logo.png'

describe('TeamInfo Component', () => {

    describe('Checking PropTypes', () => {
        it('Should NOT throw a warning', () => {
            const expectedProps = {
                teamName: jest.string,
                teamLogo: jest.string,
                voteHandler: jest.fn,
                type: jest.string,
                objectId: jest.string
            };
            const propsError = checkProps(TeamInfo, expectedProps);
            expect(propsError).toBeUndefined();
        });

    });

    describe('Component Renders', () => {
        let wrapper;
        beforeEach(() => {
            const props = {
                teamName: "Ukraine U18",
                teamLogo: HomeTeamLogo,
                voteHandler: jest.fn,
                type: "home",
                objectId: "fZZUhitsVt"
            };
            wrapper = shallow(<TeamInfo {...props} />);
        });

        it('renders correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('Should render VoteButton Component', () => {
            const desc = wrapper.find(VoteButton);
            expect(desc.length).toBe(1);
        });

    });




});