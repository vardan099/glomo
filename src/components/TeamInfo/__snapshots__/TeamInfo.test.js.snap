// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamInfo Component Component Renders renders correctly 1`] = `ShallowWrapper {}`;
